File: week_2/day_08_stack/variations_que/daily_temperatures.py
# daily_temperatures.py
def dailyTemperatures(t):
    res = [0]*len(t)
    stack = []
    for i in range(len(t)):
        while stack and t[i] > t[stack[-1]]:
            idx = stack.pop()
            res[idx] = i - idx
        stack.append(i)
    return res

# INPUT
print(dailyTemperatures([73,74,75,71,69,72,76,73]))

File: week_2/day_08_stack/variations_que/dailyTemperatures.js
// dailyTemperatures.js
function dailyTemperatures(t) {
    const res = new Array(t.length).fill(0);
    const stack = [];
    for (let i = 0; i < t.length; i++) {
        while (stack.length && t[i] > t[stack[stack.length - 1]]) {
            const idx = stack.pop();
            res[idx] = i - idx;
        }
        stack.push(i);
    }
    return res;
}

// INPUT
console.log(dailyTemperatures([73,74,75,71,69,72,76,73]));
